Kernel/Arch/riscv64/RegisterState.h

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2023, Sönke Holz <sholz8530@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <sys/arch/riscv64/regs.h>
+
+#include <Kernel/Arch/riscv64/CSR.h>
+#include <Kernel/Security/ExecutionMode.h>
+
+#include <AK/Platform.h>
+VALIDATE_IS_RISCV64()
+
+namespace Kernel {
+
+struct RegisterState {
+    u64 x[31];
+
+    u64 sstatus;
+    u64 sepc;
+    u64 scause;
+    u64 stval;
+
+    u64 user_sp;
+
+    FlatPtr userspace_sp() const { return user_sp; }
+    void set_userspace_sp(FlatPtr value) { user_sp = value; }
+
+    FlatPtr ip() const { return sepc; }
+    void set_ip(FlatPtr value) { sepc = value; }
+
+    FlatPtr bp() const { return x[7]; }
+    void set_bp(FlatPtr value) { x[7] = value; }
```

Comment:
I wouldn't mind a future refactor to call this "frame_pointer" in all the arch reigster states.